Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -13055,7 +13056,19 @@ def get_query_entries_by_id_request(job_id: str) -> Dict[str, Any]:
         Dict[str,Any]: a dictionary of the raw entries linked to the Job ID
     """
     params = assign_params(key=API_KEY, type='log', action='get', job_id=job_id)
-    return http_request(URL, 'GET', params=params)
+    
+    # if the job has not finished, wait for 1 second and try again (until success or max retries)
+    for _ in range(GET_JOB_ID_MAX_RETRIES):
+        response = http_request(URL, 'GET', params=params)
+        status = response.get('response', {}).get('result', {}).get('job', {}).get('status', '')
+        demisto.debug(f'Job ID response status: {status}')
+        if status == 'FIN':
+            demisto.debug(f'{response=}')
+            return response
+        else:
+            demisto.debug('Job not completed, Retrying in 1 second...')
+            time.sleep(1)
+    return {}
```

Comment:
```suggestion
    try_number = 1
    while try_number <= GET_LOG_JOB_ID_MAX_RETRIES:
        response = http_request(URL, 'GET', params=params)
        status = response.get('response', {}).get('result', {}).get('job', {}).get('status', '')
        demisto.debug(f'Job ID {job_id} response status: {status}, {try_number=}')
        if status == 'FIN':
            demisto.debug(f'{response=}')
            return response
        else:
            demisto.debug(f'Job {job_id} has not completed, Retrying in 1 second...')
            time.sleep(1)
        try_number += 1
    return {}
```

please extract the `job_id` as well and log it out.
i think maybe logging the response would also be good even if its not finished, would help us in the future in case of a bug.

Reply:
I added `job_id` to the log and made changes so that the response will be logged in both cases (finished or not).
Not sure how the "while" loop is better than the "for" loop in this situation...

Reply:
its just more clean instead of doing 

```
for _ in range(...)
```

in my opinion, but you don't have to accept it :)

Reply:
if log the try number then for option is the better one.

```
for try_num in range(1, GET_LOG_JOB_ID_MAX_RETRIES + 1)....
```

Reply:
Looks good, added the `try_num` to the log 👍